airflow/utils/python_virtualenv.py

```diff
@@ -35,12 +37,35 @@ def _generate_virtualenv_cmd(tmp_dir: str, python_bin: str, system_site_packages
     return cmd
 
 
-def _generate_pip_install_cmd(tmp_dir: str, requirements: List[str]) -> Optional[List[str]]:
+def _generate_pip_install_cmd(tmp_dir: str,
+                              requirements: List[str],
+                              connection_id: Optional[str] = None
+                              ) -> Optional[List[str]]:
     if not requirements:
         return None
-    # direct path alleviates need to activate
-    cmd = [f'{tmp_dir}/bin/pip', 'install']
-    return cmd + requirements
+
+    if connection_id:
+        con: Connection = BaseHook.get_connection(connection_id)
+        user = con.login
+        schema = con.schema or 'https'
+        password = con.get_password()
+        port = con.port or 8080
+        host = con.host
+        host_suffix = con.extra_dejson.get('host_suffix', 'repository/python/simple')
```

Comment:
The default seems arbitrary (specific to Artifactory?) PEP 503 does not require the index to be hosted under a suffix, so IMO this should default to an empty string. Also, `host_suffix` feels like it should be appended after `host`, not after the port part. How about calling this `path`? (This is the terminology used by `urllib.parse`.